docpipeline/generate.py

```diff
@@ -213,19 +219,48 @@ def process_blob(blob, credentials, devsite_template, xrefs):
     log.success(f"Done with {blob.name}!")
 
 
-def download_xrefs(client, bucket):
-    xrefs_dir = pathlib.Path(XREFS_DIR_NAME)
-    if xrefs_dir.is_dir():
-        shutil.rmtree(xrefs_dir)
-    xrefs_dir.mkdir(parents=True, exist_ok=True)
-    xrefs = []
-    for xref_blob in client.list_blobs(bucket, prefix=XREFS_DIR_NAME):
-        xref_path = pathlib.Path(xref_blob.name).absolute()
-        xref_path.parent.mkdir(parents=True, exist_ok=True)
-        xref_blob.download_to_filename(xref_path)
-        xrefs.append(str(xref_path))
-    log.info(f"Downloaded the xref files to {xrefs_dir.absolute()}")
-    return xrefs, xrefs_dir
+def get_xref(xref, bucket, dir):
+    if not xref.startswith(DEVSITE_SCHEME):
+        return xref
+
+    d_xref = xref[len(DEVSITE_SCHEME) :]
+    lang, pkg = d_xref.split("/", 1)
+    version = "latest"
+    if "@" in pkg:
+        pkg, version = pkg.rsplit("@", 1)
+    if version == "latest":
+        # List all blobs, sort by semver, and pick the latest.
+        prefix = f"{XREFS_DIR_NAME}/{lang}-{pkg}"
```

Comment:
Probably worth adding a - after this.

Otherwise we'll find (for example) "dotnet-Google.Cloud.Speech.V1Beta-1.0.tar.gz" when looking for "dotnet-Google.Cloud.Speech-*.tar.gz".

It would be good to add a test for this case before fixing it though :)

Reply:
Done and done. Needed a little more than a `-` because a package name can have `-` in it.